pkg/component/runtime/state.go

```diff
@@ -151,34 +145,34 @@ func (s *ComponentState) syncExpected(comp *component.Component) bool {
 		}
 
 		touched[key] = true
-		existing, ok := s.expectedUnits[key]
+		expected, ok := s.expectedUnits[key]
```

Comment:
Renaming because "existing" is also used in this file for the current units (rather than the expected units) and this makes it clearer that we're comparing to the expected config.